x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapper.java

```diff
@@ -853,6 +856,69 @@ protected String parseSourceValue(Object value) {
             };
         }
 
+        @Override
+        public QueryableExpression asQueryableExpression(SearchExecutionContext context) {
+            if (false == isSearchable()) {
+                throw new IllegalStateException("Wildcard fields are always searchable.");
+            }
+            return StringQueryableExpression.field(name(), new StringQueryableExpression.Queries() {
+                @Override
+                public Query approximateExists() {
+                    return existsQuery(context);
+                }
+
+                @Override
+                public Query approximateTermQuery(String term) {
+                    // A term query without any rechecking. The script will recheck.
+                    Set<String> ngrams = new HashSet<>();
+                    getNgramTokens(ngrams, addLineEndChars(term));
+
+                    BooleanQuery.Builder approximation = new BooleanQuery.Builder();
+                    int clauseCount = 0;
+                    for (String string : ngrams) {
+                        if (clauseCount >= MAX_CLAUSES_IN_APPROXIMATION_QUERY) {
+                            break;
+                        }
+                        addClause(string, approximation, Occur.MUST);
+                        clauseCount++;
+                    }
+                    if (clauseCount == 0) {
+                        return new MatchAllDocsQuery();
+                    }
+
+                    return approximation.build();
```

Comment:
This is quite similar to `termQuery` but it doesn't perform the double check. And it doesn't wrap through wildcards. I'm a bit scared of wrapping through wildcards here just because it seems like an opportunity for things to go wrong without providing much value. I didn't perform the double check because the script query will do that anyway. It doesn't feel worth it to do it twice.